paddle/fluid/operators/reader/create_py_reader_op.cc

```diff
@@ -14,11 +14,33 @@
 
 #include "paddle/fluid/operators/reader/lod_tensor_blocking_queue.h"
 #include "paddle/fluid/operators/reader/reader_op_registry.h"
+#include "paddle/fluid/string/split.h"
+
+DECLARE_string(selected_gpus);
 
 namespace paddle {
 namespace operators {
 namespace reader {
 
+static int GetNumPlaces(const platform::Place& place) {
+  size_t num_places = 0;
+  if (platform::is_cpu_place(place)) {
+    num_places = getenv("CPU_NUM") == NULL ? std::thread::hardware_concurrency()
+                                           : atoi(getenv("CPU_NUM"));
+  } else if (platform::is_gpu_place(place)) {
+#ifdef PADDLE_WITH_CUDA
+    num_places = FLAGS_selected_gpus.empty() == true
+                     ? platform::GetCUDADeviceCount()
+                     : paddle::string::Split(FLAGS_selected_gpus, ',').size();
+#else
+    PADDLE_THROW("PADDLE should been compiled with WITH_CUDA=1.");
+#endif
+  } else {
+    PADDLE_ENFORCE("The place of pyreader should be CPU or GPU.");
+  }
+  return num_places;
+}
```

Comment:
Maybe we should have another method to get the number of places which is used to train. In this PR(https://github.com/PaddlePaddle/Paddle/pull/15465), I add place for Parallel Executor, the above code is duplicate with the following:
https://github.com/PaddlePaddle/Paddle/blob/0c42fd826a61566444cec349e541047d1f0c4d78/python/paddle/fluid/compiler.py#L150-L190

I think that we should get the available devices for the number of devices which is used to execute in Parallel Executor, and pass the device to pyreader.